src/stim/stabilizers/pauli_string_iter.inl

```diff
@@ -0,0 +1,296 @@
+    // The above operations are simpler to implement for simd_bits as they mostly
+    // involve operations with "ones"-masks and already implemented operations like | and ^.
+    simd_bits<W> t1(cur_perm.num_bits_padded()), t2(cur_perm.num_bits_padded());
+    simd_bits<W> t3(cur_perm.num_bits_padded()), t1_inv(cur_perm.num_bits_padded());
+    uint64_t ones_bit = ~0ULL;
+    t1 = cur_perm;
+    size_t num_zeros = count_trailing_zeros(cur_perm);
+    for (size_t w = 0; w < cur_perm.num_u64_padded(); w++) {
+        // Either we have an all all ones or an all zeros integer in which case we
+        // skip, otherwise this word must contain the least significant bit
+        if (cur_perm.u64[w] != ones_bit && cur_perm.u64[w]) {
+            // Use the single word algorithm from above
+            t1.u64[w] = cur_perm.u64[w] | ((1ULL << num_zeros) - 1);
+            // Set all previous words to ones
+            for (size_t old_w = 0; old_w < w; old_w++) {
+                t1.u64[old_w] = ones_bit;
+            }
+            break;
+        }
+    }
+    // invert the bits
+    for (size_t w = 0; w < cur_perm.num_u64_padded(); w++) {
+        if (t1.u64[w]) {
+            t1_inv.u64[w] = ~t1.u64[w];
+        } else {
+            // all zeros so set to all ones
+            t1_inv.u64[w] = ones_bit;
+        }
+    }
+    size_t next_zero = count_trailing_zeros(t1_inv);
+    for (size_t w = 0; w < cur_perm.num_u64_padded(); w++) {
+        if (next_zero / 64 == w) {
+            if ((next_zero + 1) % 64 == 0) {
+                t2.u64[w] = ones_bit;
+            } else {
+                t2.u64[w] = ((1ULL << (next_zero + 1 - 64 * w)) - 1);
+            }
+            break;
+        } else {
+            t2.u64[w] = ones_bit;
+        }
+    }
+    size_t word_loc = (next_zero - num_zeros - 1) / 64;
+    for (size_t w = 0; w < word_loc; w++) {
+        t3.u64[w] = ones_bit;
+    }
+    // next_zero - num_zeros - 1 - words * 64 is always < 64, so don't need to worry about overflow like above.
+    t3.u64[word_loc] = (1ULL << (next_zero - num_zeros - 1 - word_loc * 64)) - 1;
+    cur_perm = t1;
+    cur_perm ^= t2;
+    cur_perm |= t3;
+}
+
+template <size_t W>
+bool PauliStringIterator<W>::pair_sat_increment() {
+    // This will overflow for large cur_w.
+    size_t num_terms = static_cast<size_t>(pow(3, cur_w));
+    // Set to all Xs initially
+    if (cur_k == 0) {
+        cur_k++;
+        result.xs = cur_perm;
+        result.zs.clear();
+        return true;
+    }
+    while (cur_k < num_terms) {
+        // inc = x & z
+        simd_bits<W> inc(result.xs & result.zs);
+        // up = ~inc
+        simd_bits<W> up(inc);
+        up.invert_bits();
+        // inc |= ~m
+        simd_bits<W> comp_m(cur_perm);
+        comp_m.invert_bits();
+        inc |= comp_m;
+        simd_bits<W> one(cur_perm.num_bits_padded());
+        one.u64[0] = uint64_t{1};
+        // inc += 1
+        inc += one;
```

Comment:
Maybe add a `++x` method to `simd_bits`?

Reply:
Yeah, it probably makes sense to add -- / -= too. For example, there are a few awkward parts where I'm doing (1 << (sum_number_greather_than_64) - 1.